src/Orleans/GrainDirectory/MultiClusterRegistrationStrategy.cs

```diff
@@ -8,7 +9,7 @@ namespace Orleans.GrainDirectory
     /// Strategy objects are used as keys to select the proper registrar.
     /// </summary>
     [Serializable]
-    internal abstract class MultiClusterRegistrationStrategy
+    public abstract class MultiClusterRegistrationStrategy
```

Comment:
I have looked into this a bit and I think this should be fixed by making the IProtocolServices more abstract. I will work on this.